packages/mobile/src/exchange/CeloGoldHistoryChart.tsx

```diff
@@ -0,0 +1,309 @@
+    y = y + 25
+  }
+  const [adjustedX, setAdjustedX] = useState(x)
+  const onLayout = useCallback(
+    ({
+      nativeEvent: {
+        layout: { width },
+      },
+    }: LayoutChangeEvent) => {
+      if (width !== chartWidth) {
+        if (x - width / 2 < 0) {
+          setAdjustedX(width / 2)
+        }
+        if (x + width / 2 > chartWidth) {
+          setAdjustedX(chartWidth - width / 2)
+        }
+      }
+    },
+    [x]
+  )
+
+  return (
+    <SvgText onLayout={onLayout} fill="black" fontSize="14" x={adjustedX} y={y} textAnchor="middle">
+      {value}
+    </SvgText>
+  )
+}
+
+function renderDotOnChart(chartData: Array<{ amount: number | BigNumber; displayValue: string }>) {
+  let lowestRateIdx = 0,
+    highestRateIdx = 0
+  chartData.forEach((rate, idx) => {
+    if (rate.amount > chartData[highestRateIdx].amount) {
+      highestRateIdx = idx
+    }
+    if (rate.amount < chartData[lowestRateIdx].amount) {
+      lowestRateIdx = idx
+    }
+  })
+  return ({ x, y, index }: { x: number; y: number; index: number }) => {
+    switch (index) {
+      case chartData.length - 1:
+        return (
+          <G key={index}>
+            <Circle cx={x} cy={y} r="5" fill={'black'} />
+            <Line x1={0} y1={y} x2={x} y2={y} stroke={colors.listBorder} strokeWidth="1" />
+          </G>
+        )
+
+      case highestRateIdx:
+        return (
+          <ChartAwareSvgText
+            x={x}
+            y={y}
+            key={index}
+            value={chartData[highestRateIdx].displayValue}
+            position={'top'}
+            chartWidth={CHART_WIDTH}
+          />
+        )
+
+      case lowestRateIdx:
+        return (
+          <ChartAwareSvgText
+            x={x}
+            y={y}
+            key={index}
+            value={chartData[lowestRateIdx].displayValue}
+            position={'bottom'}
+            chartWidth={CHART_WIDTH}
+          />
+        )
+
+      default:
+        return null
+    }
+  }
+}
+
+function Loader() {
+  return (
+    <View style={styles.loader}>
+      <ActivityIndicator size="large" color={colors.celoGreen} />
+    </View>
+  )
+}
+
+function CeloGoldHistoryChart({ t, testID }: Props) {
+  const dispatch = useDispatch()
+  const calculateGroup = useCallback((er) => {
+    return Math.floor(er.timestamp / (range / CHART_POINTS_NUMBER))
+  }, [])
+
+  const localCurrencyCode = useLocalCurrencyCode()
+  const displayLocalCurrency = useCallback(
+    (amount: BigNumber.Value) =>
+      getLocalCurrencyDisplayValue(amount, localCurrencyCode || LocalCurrencyCode.USD, true),
+    [localCurrencyCode]
+  )
+  const goldToDollars = useCallback(useGoldToDollarAmount, [])
+  const dollarsToLocal = useCallback(useDollarsToLocalAmount, [])
+
+  const [range] = useState(30 * 24 * 60 * 60 * 1000) // 30 days
+  useEffect(() => {
+    const interval = setInterval(() => {
+      dispatch(syncCeloGoldExchangeRateHistory())
+    }, 600000) // Sync history every 10 minutes
```

Comment:
long intervals are discouraged in React Native. Can we avoid this somehow? Or just rely on firebase to trigger when the data changes?

Reply:
I would take a look on how to avoid it using not `once` as you mentioned below! Thanks!